### ISSUE TITLE
Query: How do people run pipelines concurrently with defined PVC workspaces?

### ISSUE THREAD

**Issue by chabeezy** [tektoncd/pipeline#5412]
Hi,

First of all great work on tekton, it's a great product! 

I was just curious, and I've struggled to find out a way to do this. I want to be able to run multiple pipelines at once, our use case is we run a set of tests on PR's when they are created/updated. It's possible for people to be pushing to different PR's at the same time. Thus triggering pipelines to run at the same time

Currently the pipeline queues up, I imagine this is because the workspace is in use. 

Just wondering if its possible to give a selection of workspaces, or something similar? As in a pipeline has workspaces x,y and z. If x and y are in use, use z?

I've read you can use templates that will be deleted if you delete the pipeline run.. however I think we need to hold onto the pipeline runs for too long for that to work for us. Can you tear them down at the end of a pipeline run?

**Comment by jerop** [comment]
> Just wondering if its possible to give a selection of workspaces, or something similar? As in a pipeline has workspaces x,y and z. If x and y are in use, use z?

@chabeezy wondering if you have a selection of pre-defined workspaces only because they aren't be torn down when the pipelinerun are completed...or if there are other reasons

**Comment by chabeezy** [comment]
Yes currently we don’t tear down the pipelines after each run, as we need to preserve the logs for a while. 

We don’t have lots of space either so can’t create pvc templates as we’ll soon run out, or lose logs too soon due to deleting the pipeline run. 

I’ve not implemented it yet, but my plan is

- Send the logs to elasticsearch 
- I can then use a template for pvc and implement the cron job that deletes pipeline runs periodically.

**Comment by jerop** [comment]
@chabeezy you may be interested in the work that @adambkaplan is doing to persist logs in tekton - https://github.com/tektoncd/community/blob/main/teps/0117-tekton-results-logs.md

**Comment by austinzhao-go** [comment]
thanks @jerop for linking my question to this discussion.

perhaps a related use case from my problem
```
Use Case
need a pvc to pass fetched secrets between Tasks in a PipelineRun

Expected:
pvc will be deleted upon the completion of PipelineRun as "grinding" the fetched creds   
Current:
pvc will be kept and only deleted when deleting the TaskRun/PipelineRun files
```

thank you @chabeezy for bringing up this FR discussion!

**Comment by jerop** [comment]
/reopen
/lifecycle frozen

issue is not addressed

**Comment by pritidesai** [comment]
> thanks @jerop for linking my question to this discussion.
> 
> perhaps a related use case from my problem
> 
> ```
> Use Case
> need a pvc to pass fetched secrets between Tasks in a PipelineRun
> 
> Expected:
> pvc will be deleted upon the completion of PipelineRun as "grinding" the fetched creds   
> Current:
> pvc will be kept and only deleted when deleting the TaskRun/PipelineRun files
> ```
> 
> thank you @chabeezy for bringing up this FR discussion!

@austinzhao-go I have the PoC created here - https://github.com/tektoncd/pipeline/pull/6635 Please let me know if this works for you. Thanks!

**Comment by dcheng1248** [comment]
Hi, I'm wondering what is the state of this issue? I would really appreciate being able to have automatically deleted PVCs tied to each pipeline run so my pipelines don't hang because they can't get access to a RWO PVC. A RWX PVC causes various errors for my pipelines. 

From looking at the various issues, it looks like the automatic PVC deletion is implemeneted for the coschedule: pipelinerun and coschedule: isolate-pipelinerun modes but not for the coschedule: workspace mode (are there plans to implement this or is that not planned)? Additionally, I may not be able to change the cluster-wide Tekton configuration, so I am unsure if I could make use of the newer affinity modes or the affinity assistant themselves. Any pointers on the issue status or potential workarounds would be appreciated. I would also be happy to look into the source code and contribute if it helps with the implementation. Thanks!
